### PR TITLE
Refactor: enforce uniform database naming schema

### LINKED ISSUE #3480: Ensure we have a uniform database-column naming schema through the project
We currently have a mixed schema in different models. They need to be unified such that it's easier to under stand and maintain. 

| Component                       | Naming style (`snake_case`)                                                                                                                                                                                                                                        | Example                                                                                                                                                                                                                                                                                    |
|---------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Table                           | plural                                                                                                                                                                                                                                                             | `posts`, `sessions`, `events`, `speakers`                                                                                                                                                                                                                                                   |
| Model file                      | `<singular of table name>.py`                                                                                                                                                                                                                                      | `event.py`, `session.py`                                                                                                                                                                                                                                                                   |
| Column                          | 1. `<name>` - singular if it represent a single data. Else plural. <br>   2. If date only column: `<name>_date`<br>  3. If date-time column: `<name>s_at` <br> 4. If the column contains an image url, or any type of url, it should be suffixed with `_url` <br> 5. If boolean field: `is_<name>` | 1. `name`, `description` <br> 2. `start_date`, `end_date` <br> 3. `starts_at`, `ends_at` <br> 4. `image_url`, `logo_url`, `event_url`, `ticket_url` <br> 5. `is_active`, `is_admin`, `is_verified`                                                                                                                            |
| Foreign Key id column           | `<singular of table name>_id`                                                                                                                                                                                                                                      | `session_id`, `speaker_id`                                                                                                                                                                                                                                                                 |
| Relationship                    | 1. `<name>` - singular for one-to-one<br> 2. `<name>` - plural for one-to-many<br> 3. `<name>` - singular for many-to-one<br> 3. `<name>` - plural for many-to-many<br>                                                                                          | 1. An event model can have a `call_for_speakers` relation <br> 2. Each track model will have a `sessions` relation <br> 3. Each session model will have a `track` relation <br> 4. Each session model will have a `speakers` relation. Each speaker model will have a `sessions` relation. |
| many-to-many intermediate table | The name is derived from the alphabetical order of the related model names.  `<name_of_model_one>_<name_of_model_two>` (model one and two in alphabetical order)                                                                                                   | A table that holds the relation between events and users will be called.  `event_user`                                                                                                                                                                                                     |

The above table will also be added to the project docs for future reference.

This is a parent issue. Sub-issues can be created.

### PR DESCRIPTION
**Pull request by shubham-padia**
fixes #3480 .
Please review   :shipit: 
If you find any discrepancies with the naming schema that were missed during the refactor, please comment here or open an issue later when you notice it.

### CODE REVIEW

File: app/helpers/wizard/event.py [hunk 1]
```diff
@@ -326,7 +326,7 @@ def save_tickets(tickets_data, event):
         ticket.max_order = ticket_data['max_order'] if ticket_data['max_order'] != '' else 10
         ticket.sales_start = get_event_time_field_format(ticket_data, 'sales_start')
         ticket.sales_end = get_event_time_field_format(ticket_data, 'sales_end')
-        ticket.absorb_fees = ticket_data['absorb_fees']
+        ticket.is_absorbed_fees = ticket_data['is_absorbed_fees']
```
Comment (niranjan94) [inline app/helpers/wizard/event.py]: let this be call `is_fee_absorbed`

File: app/models/session.py [hunk 2]
```diff
@@ -95,7 +95,7 @@ def __init__(self,
         self.level = level
         self.created_at = created_at
         self.deleted_at = deleted_at
-        self.state_email_sent = state_email_sent
+        self.is_sent_email = is_sent_email
```
Comment (niranjan94) [inline app/models/session.py]: rename to `is_mail_sent`

File: app/models/tax.py [hunk 3]
```diff
@@ -31,27 +31,27 @@ def __init__(self,
                  tax_name=None,
                  tax_rate=None,
                  tax_id=None,
-                 send_invoice=None,
+                 is_sent_invoice=None,
```
Comment (niranjan94) [inline app/models/tax.py]: rename to `is_invoice_sent`

File: app/models/ticket.py [hunk 4]
```diff
@@ -18,7 +18,7 @@ class Ticket(db.Model):
     type = db.Column(db.String)
     quantity = db.Column(db.Integer)
     price = db.Column(db.Float)
-    absorb_fees = db.Column(db.Boolean)
+    is_absorbed_fees = db.Column(db.Boolean)
```
Comment (niranjan94) [inline app/models/ticket.py]: rename to `is_fee_absorbed`

File: app/static/js/events/wizard/basic-details/vue-models.js [hunk 5]
```diff
@@ -60,14 +60,14 @@ var EVENT = {
         tax_name: '',
         tax_rate: '',
         tax_id: '',
-        send_invoice: false,
+        is_sent_invoice: false,
```
Comment (niranjan94) [inline app/static/js/events/wizard/basic-details/vue-models.js]: rename to `is_invoice_sent`

File: app/static/js/events/wizard/basic-details/vue-models.js [hunk 6]
```diff
@@ -98,5 +98,5 @@ var TICKET = {
     min_order: 1,
     max_order: 10,
     tags_string: '',
-    absorb_fees: false
+    is_absorbed_fees: false
```
Comment (niranjan94) [inline app/static/js/events/wizard/basic-details/vue-models.js]: Rename to `is_fee_absorbed`

**Comment by shubham-padia** [comment]
@niranjan94 Changed made as requested.

**Comment by shubham-padia** [comment]
@niranjan94 Should I open an issue about adding tests for checking this in the future ?

**Comment by niranjan94** [comment]
@shubham-padia yes. Please open an issue

**Comment by mariobehling** [comment]
We have the OpenTechSummit on May 25. So, maybe merge after that.

**Comment by SaptakS** [comment]
@shubham-padia this Pull Request also has merge conflicts. You can solve them after May 25th if you want because there might be many more conflicts.

**Pull request closed without merging**